k8s-openapi-codegen-common/src/lib.rs

```diff
@@ -1893,3 +1916,12 @@ fn get_operation_names(
 
 	Ok((operation_fn_name, operation_result_name, operation_optional_parameters_name))
 }
+
+/// Ensure a field name is safe to be used as Rust identifier
+fn safe_field(name: &str) -> String {
```

Comment:
Merge this into `get_rust_ident`

Reply:
The problem with that is that `get_rust_ident` will be used for field names where this is only a part of the field name, e.g. like:

~~~
let mut value_{field_name} = {field_name}
~~~

Now this gets expanded into:

~~~
let mut value_as = r#as;
~~~

Merging that into `get_rust_ident` will render:

~~~
let mut value_r#as = r#as;
~~~

And I am not sure that works.

Reply:
Yes, `value_r#as` won't work. The renaming that `get_rust_ident` does doesn't use raw literals, especially since it predates edition 2018, so this has never been a problem. There's two ways to resolve this:

1. Map `as` to `as_` instead of `r#as`. This will be more consistent with the existing mappings like `"type"` to `"type_"` anyway.

1. Have `get_ident` return two strings, one that is the existing one and is suitable as a standalone ident, and one that is meant to be suffixed to other idents. The former can then be the `r#` form and used as the `field_name`, and the latter can be the non-`r#` form and used for composite idents like `Key_{}` and `value_{}`.

Reply:
So I went ahead and merged the two functions, adopting the `_` suffix pattern that you already had. I still think it would be nicer to use Rust's raw identifiers in this case, but this isn't something for this PR I think.

Reply:
Sorry, I didn't see your comment before, while writing my own :grinning: …  I think this should be fixed now as described in your first option.